src/select/__tests__/select.test.js

```diff
@@ -71,4 +71,19 @@ describe('Select component', function() {
       value: [],
     });
   });
+  test('select flow allows custom keys in options objects', function() {
+    wrapper = mount(
+      <Select
+        options={[
+          {id: 'AliceBlue', color: '#F0F8FF'},
+          {id: 'AntiqueWhite', color: '#FAEBD7'},
+        ]}
+        closeOnSelect={false}
+        onChange={({option}) => option.color}
```

Comment:
You'll want to add a `$FlowFixMe` above lines that we expect to fail the `onChange` call back should look something like below.

```js
onChange={({option}) => {
  // accessing items that may not exist on option continues to fail
  // $FlowFixMe
  console.log(option.color);

  if (option.color) {
    // it works when we assert its existence
    console.log(option.color);
  }
}}
```

Reply:
I tried
```
test('select flow allows custom keys in options objects', function() {
    wrapper = mount(
      <Select
        options={[
          {id: 'AliceBlue', color: '#F0F8FF'},
          {id: 'AntiqueWhite', color: '#FAEBD7'},
        ]}
        closeOnSelect={false}
        onChange={({option}) => {
          /* eslint-disable no-console */
          // $FlowFixMe
          console.info(option.color);
          if (option.color) {
            console.info(option.color);
          }
          /* eslint-enable no-console */
        }}
        labelKey="id"
        multi
        valueKey="color"
      />,
    );
  });
```
But this still throws a `flow` error locally because `color` is not defined.

Reply:
which line gives that error? it's probably because we don't know what the type of `color` is so cannot be safely provided to `console.info`

Reply:
or needs to be more exhaustively checked

`if (option.color !== null && option.color !== undefined)`

Reply:
Fixed with `if (option !== null && option !== undefined && option.color)` 👍